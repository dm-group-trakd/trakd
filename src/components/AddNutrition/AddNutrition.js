import React, { Component } from 'react';
import { Card, CardContent } from '@material-ui/core';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import axios from 'axios';
import './styles/AddNutrition.scss';


class AddNutrition extends Component {
    constructor(){
        super();
        this.state = {
            searchedFood: '',
            searchResults1: [],
            searchResults2: [],
            searchResults3: []
        }
    }
    
    handleInput = e => {
        this.setState({
            [e.target.name]: e.target.value,
            searchResults1: [],
            searchResults2: [],
            searchResults3: [],
            searchResults4: [],
            searchResults5: [],
            searchResults6: [],
            searchResults7: [],
            searchResults8: [],
            searchResults9: [],
            searchResults10: []
        })

    }
    
    handleSearchSubmit = () => {
        axios.get(`https://api.edamam.com/api/food-database/parser?ingr=${this.state.searchedFood}&app_id=${process.env.REACT_APP_app_id}&app_key=${process.env.REACT_APP_app_key}`)
        .then(res => {
            console.log(res.data.hints[0])
            this.setState({
                searchResults1: [...this.state.searchResults1, res.data.hints[0].food],
                searchResults2: [...this.state.searchResults1, res.data.hints[1].food],
                searchResults3: [...this.state.searchResults1, res.data.hints[2].food],
                searchResults4: [...this.state.searchResults1, res.data.hints[3].food],
                searchResults5: [...this.state.searchResults1, res.data.hints[4].food],
                searchResults6: [...this.state.searchResults1, res.data.hints[5].food],
                searchResults7: [...this.state.searchResults1, res.data.hints[6].food],
                searchResults8: [...this.state.searchResults1, res.data.hints[7].food],
                searchResults9: [...this.state.searchResults1, res.data.hints[8].food],
                searchResults10: [...this.state.searchResults1, res.data.hints[9].food]
            })
            console.log(this.state.searchResults1)
        })
    }

    render() {
        return (
            <div className="add-nutrition-main">
                <Card >
                    <div className="card-content">
                    <CardContent>
                        <h1>Add Food</h1>
                        <TextField 
                            id="outlined-Password-input"
                            label="Search"
                            type="text"
                            name="searchedFood"
                            autoComplete="search"
                            margin="normal"
                            variant="outlined"
                            onChange={this.handleInput}
                        />
                        <Button 
                            variant="contained" 
                            color="primary"
                            onClick={this.handleSearchSubmit}
                            className = "search-button"
                        >Search</Button>
                        {this.state.searchResults1 ? this.state.searchResults1.map((food, i) => (
                            <div className="search-results" key={i}>
                                <h1>{food.label}</h1>
                                <h1>{food.nutrients.ENERC_KCAL}</h1>
                            </div>
                        )): null}
                        {this.state.searchResults2 ? this.state.searchResults2.map((food, i) => (
                            <div className="search-results" key={i}>
                                <h1>{food.label}</h1>
                                <h1>{food.nutrients.ENERC_KCAL}</h1>
                            </div>
                        )): null}
                        {this.state.searchResults3 ? this.state.searchResults3.map((food, i) => (
                            <div className="search-results" key={i}>
                                <h1>{food.label}</h1>
                                <h1>{food.nutrients.ENERC_KCAL}</h1>
                            </div>
                        )): null}
                        {this.state.searchResults4 ? this.state.searchResults4.map((food, i) => (
                            <div className="search-results" key={i}>
                                <h1>{food.label}</h1>
                                <h1>{food.nutrients.ENERC_KCAL}</h1>
                            </div>
                        )): null}
                        {this.state.searchResults5 ? this.state.searchResults5.map((food, i) => (
                            <div className="search-results" key={i}>
                                <h1>{food.label}</h1>
                                <h1>{food.nutrients.ENERC_KCAL}</h1>
                            </div>
                        )): null}
                        {this.state.searchResults6 ? this.state.searchResults6.map((food, i) => (
                            <div className="search-results" key={i}>
                                <h1>{food.label}</h1>
                                <h1>{food.nutrients.ENERC_KCAL}</h1>
                            </div>
                        )): null}
                        {this.state.searchResults7 ? this.state.searchResults7.map((food, i) => (
                            <div className="search-results" key={i}>
                                <h1>{food.label}</h1>
                                <h1>{food.nutrients.ENERC_KCAL}</h1>
                            </div>
                        )): null}
                        {this.state.searchResults8 ? this.state.searchResults8.map((food, i) => (
                            <div className="search-results" key={i}>
                                <h1>{food.label}</h1>
                                <h1>{food.nutrients.ENERC_KCAL}</h1>
                            </div>
                        )): null}
                        {this.state.searchResults9 ? this.state.searchResults9.map((food, i) => (
                            <div className="search-results" key={i}>
                                <h1>{food.label}</h1>
                                <h1>{food.nutrients.ENERC_KCAL}</h1>
                            </div>
                        )): null}
                        {this.state.searchResults10 ? this.state.searchResults10.map((food, i) => (
                            <div className="search-results" key={i}>
                                <h1>{food.label}</h1>
                                <h1>{food.nutrients.ENERC_KCAL}</h1>
                                <Button>Add</Button>
                            </div>
                        )): null}
                    </CardContent>
                    </div>
                </Card>
                
            </div>
        )
    }
}

export default AddNutrition
