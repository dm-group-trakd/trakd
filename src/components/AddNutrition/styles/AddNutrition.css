.search-mapped {
  max-height: 50vh;
  max-width: 60vw;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -ms-flex-wrap: wrap;
      flex-wrap: wrap;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: space-evenly;
      -ms-flex-pack: space-evenly;
          justify-content: space-evenly;
  overflow-y: scroll;
  margin-top: 2vh;
}

.search-results-p {
  width: 100%;
  height: 100px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-shadow: -2px 11px 21px -16px rgba(0, 0, 0, 0.75);
          box-shadow: -2px 11px 21px -16px rgba(0, 0, 0, 0.75);
  margin: 10px 10px 10px 10px;
  -webkit-box-pack: space-evenly;
      -ms-flex-pack: space-evenly;
          justify-content: space-evenly;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
}

.search-results-p h2 {
  font-size: 1.25rem;
  font-weight: 600;
}

.search-results {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: space-evenly;
      -ms-flex-pack: space-evenly;
          justify-content: space-evenly;
}
/*# sourceMappingURL=AddNutrition.css.map */