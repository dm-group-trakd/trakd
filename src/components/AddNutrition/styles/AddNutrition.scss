.search-mapped{
    max-height: 50vh;
    max-width: 60vw;
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    justify-content: space-evenly;
    overflow-y: scroll;
    margin-top: 2vh;
}

.search-results-p{
    width: 100%;
    height: 100px;
    display: flex;
    flex-direction: column;
    box-shadow: -2px 11px 21px -16px rgba(0,0,0,0.75);
    margin: 10px 10px 10px 10px;
    justify-content: space-evenly;
    align-items: center;
}
.search-results-p h2{
    font-size: 1.25rem;
    font-weight: 600;
}

.search-results {
    display: flex;
    width: 100%;
    justify-content: space-evenly;
}