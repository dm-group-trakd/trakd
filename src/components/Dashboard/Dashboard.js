import React, { Component } from "react";
import Calories from "../calories/Calories"
import Food from "../Food/Food"
import Tips from "../tips/Tips"
import "./styles/Dashboard.css"
import axios from 'axios'

export default class Dashboard extends Component {
    constructor() {
        super();
        this.state = {
           tips:""
        }
    }
   
    render() {
        return (
            <div className="dashboard-container">
                {/* <h1>Dashboard</h1> */}
                <div className ="right-components">
                    <Calories />
                <div>
                    <Tips
                    />
                </div>
                </div>
                <div>
                    <Food />
                </div>
            </div>
        )
    }

}