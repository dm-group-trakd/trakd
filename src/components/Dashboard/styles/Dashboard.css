.dashboard-container {
  width: 90vw;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
      -ms-flex-pack: space-evenly;
          justify-content: space-evenly;
}

.dash-components {
  width: 90%;
  height: 80vh;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -ms-flex-pack: distribute;
      justify-content: space-around;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
      -ms-flex-direction: row;
          flex-direction: row;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
}

.right-components {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
      -ms-flex-pack: justify;
          justify-content: space-between;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
}

#Arrows {
  width: 120px;
}

@media (max-width: 1600px) {
  #Arrows {
    display: none;
  }
}

@media (max-width: 1500px) {
  .dash-components {
    -webkit-box-pack: center;
        -ms-flex-pack: center;
            justify-content: center;
    -webkit-box-align: none;
        -ms-flex-align: none;
            align-items: none;
    -ms-flex-wrap: wrap;
        flex-wrap: wrap;
  }
  .right-components {
    -webkit-box-align: center;
        -ms-flex-align: center;
            align-items: center;
    -webkit-box-pack: center;
        -ms-flex-pack: center;
            justify-content: center;
  }
  .dashboard-container {
    -webkit-box-pack: center;
        -ms-flex-pack: center;
            justify-content: center;
    -webkit-box-align: center;
        -ms-flex-align: center;
            align-items: center;
  }
}

@media (max-width: 800px) {
  .dashboard-container {
    margin-top: 10vh;
  }
}
/*# sourceMappingURL=Dashboard.css.map */