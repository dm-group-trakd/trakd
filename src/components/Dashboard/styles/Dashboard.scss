.dashboard-container {
    display: flex;
    justify-content: space-around;
    width:100vw;
}
.right-components{
    display:flex;
    flex-direction: column;
    align-items: center;
}