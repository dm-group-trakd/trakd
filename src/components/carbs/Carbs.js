import React from 'react'
import './styles/carbs.scss'
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import {Doughnut } from 'react-chartjs-2'
import {connect} from 'react-redux'
import {getFood} from '../../redux/reducers/foodReducer'
import {getGoals} from '../../redux/reducers/userReducer'
import Button from '@material-ui/core/Button';

class Carbs extends React.Component {
    constructor() {
        super()
        this.state = {
            eaten: 0
        }
    }

    componentDidMount = () => {
        this.props.getFood().then(() => {

            let carbsTotal = 0
            for (let i = 0; i < this.props.food.length; i++) {
                carbsTotal += this.props.food[i].carbs
            }
            this.setState({eaten: carbsTotal})
        })
        this.props.getGoals()
    }

    componentDidUpdate(prevProps) {
        if(JSON.stringify(prevProps.food) !== JSON.stringify(this.props.food)) {
            this.props.getFood().then(() => {
                let carbsTotal = 0
                for (let i = 0; i < this.props.food.length; i++) {
                    carbsTotal += this.props.food[i].carbs
                }
                this.setState({
                        eaten: carbsTotal   
                })
            })
        }
    }

    render() {
        const data = {
            labels: ["", "Carbs"],
            datasets: [
                {
                    data: [this.state.eaten, this.props.carbs_goal - this.state.eaten ],
                    backgroundColor: ["#36A2EB", "#dae0e6"]
                }
            ]
        };
        return (
            <div className='calorie-container'>
                
                <Card>
                <h1 id="Nutrient-Header">Carbs</h1>
                <div className="Nutrients-Nav">
                <Button onClick={this.props.showCalories}  color="primary"  variant="contained" >Calories</Button>
                <Button onClick = {this.props.showProtein}  color="primary"  variant="contained" >Protein</Button>
                <Button onClick = {this.props.showFat}  color="primary"  variant="contained" >Fat</Button>
                
                
                </div>
                    <CardContent>
                        <Doughnut data={data} width="400" height="300" options={{responsive: true}} />
                    </CardContent>
                </Card>
            </div>
        )
    }
}
const mapStateToProps = reduxState => {
    return {
        food: reduxState.foodReducer.food,
        carbs_goal: reduxState.userReducer.carbs_goal
    }
}
export default connect(mapStateToProps, {
getFood,
getGoals
})(Carbs);