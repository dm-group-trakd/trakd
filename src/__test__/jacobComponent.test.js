import React from 'react'
// import ReactDOM from 'react-dom';
// import {act, Simulate} from 'react-dom/test-utils';
import Enzyme, {shallow, mount, unmount} from "enzyme"
import {Provider} from "react-redux"
import configureMockStore from "redux-mock-store"
import Carbs from '../components/carbs/Carbs';
import Dashboard from '../components/Dashboard/Dashboard';
import AddNutrition from '../components/AddNutrition/AddNutrition';
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({adapter: new Adapter() })

const mockStore = configureMockStore();
const store = mockStore({});

describe("Fat Component", () => {
    test("should render without throwing an error", () => {
        expect(
            shallow(
                <Provider store={store}>
                    <Carbs />
                </Provider>
            ).exists()
        ).toBe(true);
    });
});

describe("Dashboard Component", () => {
    test("should render without throwing an error", () => {
        expect(
            shallow(
                <Provider store={store}>
                    <Dashboard />
                </Provider>
            ).exists()
        ).toBe(true);
    });
});

describe("AddNutrition Component", () => {
    test("should render without throwing an error", () => {
        expect(
            shallow(
                <Provider store={store}>
                    <AddNutrition />
                </Provider>
            ).exists()
        ).toBe(true);
    });
});

describe('AddNutrition', () => {
        it('activate button with click for handle-search button', () => {
            const component = mount(<AddNutrition.WrappedComponent />);
            component
                .find('button#handle-search')
                .simulate('click', null);
            expect(component).toMatchSnapshot();
            component.unmount();
        });
        
        it('setState of searchedFood when value ', () => {
            const component = mount(<AddNutrition.WrappedComponent />);
            component
                .find('input#outlined-Search-Food-input')
                .simulate('change', {target: {value: 'Test'}});
            expect.setState({searchedFood: 'Test'});
            component.unmount();
        });
});

// let container = null;

// beforeEach(()=> {
//     container = document.createElement('div');
//     document.body.appendChild(container);
// })

// afterEach(()=> {
//     ReactDOM.unmountComponentAtNode(container);
//     document.body.removeChild(container);
//     container = null;
// })

// test(`StudentQuestions should render properly`, () => {
//     act(() => {
//         ReactDOM.render(<AddNutrition.WrappedComponent />, container)
//     })

//     const firstH1 = document.querySelector('h1:nth-child(1)');
//     expect(firstH1.textContent).toBe('Add Food')
// })

