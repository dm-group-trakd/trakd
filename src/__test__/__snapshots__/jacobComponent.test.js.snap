// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNutrition activate button with click for handle-search button 1`] = `ReactWrapper {}`;

exports[`activate button with click for handle-search button 1`] = `ReactWrapper {}`;

exports[`should be possible to activate button with click 1`] = `ReactWrapper {}`;
